zap/src/main/java/org/parosproxy/paros/control/Control.java

```diff
@@ -139,6 +143,24 @@ private Control() {
     }
 
     private boolean init(ControlOverrides overrides, boolean startProxy) {
+        AddOnLoader addOnLoader =
+                ExtensionFactory.getAddOnLoader(
+                        model.getOptionsParam().getCheckForUpdatesParam().getAddonDirectories());
+        if (overrides != null) {
+            AddOnCollection addOnCollection = addOnLoader.getAddOnCollection();
+            overrides
+                    .getRequiredAddOns()
+                    .forEach(
+                            id -> {
+                                AddOn addOn = addOnCollection.getAddOn(id);
+                                if (addOn == null) {
+                                    String message = "The required add-on was not found: " + id;
```

Comment:
Maybe add: "If you are building ZAP from the source code then you will need to install the add-on from zap-extensions". Could we have a new task in zap-extensions, like "install-essential"? Or would that be too tricky? ... I guess we could still give the instructions like: "In zap-extensions call 'gradle ...' with the right add-on name...

Reply:
I'd prefer update the dev docs instead of hardcode instructions.

No, it's easy to add such task, but we'll need to maintain the list of required add-ons in two places.

Reply:
Could we link to the dev docs then? I agree they are better than hardcoded instructions

Reply:
OK.

Reply:
Tweaked the message.